R/tidyeval.R

```diff
@@ -40,6 +40,37 @@ capture_dots <- function(.data, ..., .j = TRUE) {
   unlist(dots, recursive = FALSE)
 }
 
+capture_new_vars <- function(.data, ...) {
+  dots <- as.list(enquos(..., .named = TRUE))
+  for (i in seq_along(dots)) {
+    dot <- dt_squash(dots[[i]], data = .data) 
+    if (!is.null(dot)) dots[[i]] <- dot
+    .data$vars <- union(.data$vars, names(dots)[i])
+  }
+
+  # Remove names from any list elements
+  is_list <- vapply(dots, is.list, logical(1))
+  names(dots)[is_list] <- ""
+
+  # Auto-splice list results from dt_squash()
+  dots[!is_list] <- lapply(dots[!is_list], list)
+  out <- unlist(dots, recursive = FALSE)
+
+  # detect temp vars i.e. repeated vars where last is var = NULL
+  var_is_null <- vapply(out, identical, y = quo(NULL), lgl(1))
+  if (any(var_is_null)) {
+    var_is_last <- rev(data.table::rowid(rev(names(out))) == 1)
```

Comment:
Another option is below. Neither of these seem good.
`!duplicated(match(names(out), rev(names(out))), fromLast = TRUE)`